### PR TITLE
feature-benchmark: Separate clusterd and measure memory separately

### PR DESCRIPTION
**Pull request by def-**
As requested in https://materializeinc.slack.com/archives/CM7ATT65S/p1717739202702299?thread_ts=1717737821.704999&cid=CM7ATT65S

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by def-** [comment]
I'll test if this works as expected in https://github.com/MaterializeInc/materialize/pull/27485 (locally looked good so far)
@nrainer-materialize What is the migration strategy for adding columns to the feature-bench result uploads? I assume I should bump a version somewhere and set a DDL for how to migrate from the old version to current. Is that supported?

**Comment by nrainer-materialize** [comment]
> I'll test if this works as expected in #27485 (locally looked good so far) @nrainer-materialize What is the migration strategy for adding columns to the feature-bench result uploads? I assume I should bump a version somewhere and set a DDL for how to migrate from the old version to current. Is that supported?

This is rather manual at the moment. Steps are:
* Increase the data version in `TEST_ANALYTICS_DATA_VERSION` in `misc/python/materialize/test_analytics/test_analytics_db.py`
* Adjust the setup scripts in `misc/python/materialize/test_analytics/setup/structures`
* Change the structures in the database
  *  drop the changed table (it is not possible to alter columns at the moment; it is ok for me at the moment to still loose data), if needed also with cascade, manually run the new script
  * This can be done before merging the PR; we should sync on this so that we do not overwrite each others changes.
  * Changes should be backwards compatible if possible (in order not to break other branches on earlier code states). In particular, newly added columns should not be `NOT NULL` (or specify a `DEFAULT`).
* Run `GRANT SELECT, INSERT, UPDATE ON ALL TABLES IN DATABASE test_analytics TO "infra+qacanaryload@materialize.io";` (specified in `misc/python/materialize/test_analytics/setup/structures/00-database.sql`) at the end to grant access to the re-created table
* Adjust insert code in `misc/python/materialize/test_analytics/data/feature_benchmark/feature_benchmark_result_storage.py`

In future, we might want to automate this to some degree and use migrations (copying data to a temporary table when replacing a table to change its structure).

Reach out if anything is unclear.

**Comment by nrainer-materialize** [comment]
Though, for this change and given the early state, feel free to break the data structures. Upload failures no longer cause builds to fail on all branches that are based on https://github.com/MaterializeInc/materialize/pull/27492.

File: misc/python/materialize/feature_benchmark/benchmark.py
Comment (nrainer-materialize) [inline misc/python/materialize/feature_benchmark/benchmark.py:185]: Optional: This snippet is rather similar to the previous one. Can we extract a function?

File: misc/python/materialize/feature_benchmark/benchmark_versioning.py
Comment (nrainer-materialize) [inline misc/python/materialize/feature_benchmark/benchmark_versioning.py:33]: Sorry 🙈 
We will discuss this next Tuesday and either get rid of it or at least reduce the number of hashes.

**Review by nrainer-materialize — APPROVED** [review]
This is great!

File: misc/python/materialize/feature_benchmark/benchmark.py
Comment (def-) [reply misc/python/materialize/feature_benchmark/benchmark.py:185]: I thought about that too, but probably increases the complexity more than it helps.

File: misc/python/materialize/test_analytics/data/feature_benchmark/feature_benchmark_result_storage.py
Comment (nrainer-materialize) [inline misc/python/materialize/test_analytics/data/feature_benchmark/feature_benchmark_result_storage.py:68]: I think a comma is missing here ⚠️

**Comment by benesch** [comment]
Awesome, thank you! Don’t have time to review, so don’t block on my approval!

**Pull request merged**
